Give tiny planet make the client disconect from server

### Your GTNH Discord Username

mamiemru

### Your Pack Version

daily 2026 1 10

### Your Server

MP

### Java Version

Java 25

### Type of Server

None

### Your Expectation

Hey.
Enable cheat mode in nei, type any EOH planet, take some.

The client is disconnected, the server print error (see below), you can reconnect and have your planet in the inv.

### The Reality

[22:01:24] [Server thread/INFO] [EnderCore]: Sending server configs to client for com.enderio.core.common.config.ConfigHandler
[22:01:24] [Server thread/INFO] [mobsinfo]: Sending config to mamiemru
[22:01:24] [Server thread/INFO] [kubatech]: Sending config to mamiemru
[22:01:24] [Server thread/INFO] [EnderCore]: Sending server configs to client for tterrag.wailaplugins.config.WPConfigHandler
[22:01:24] [Server thread/INFO] [Waila]: Player EntityPlayerMP['mamiemru'/134520040, l='World', x=200.11, y=122.53, z=117.68] connected. Sending ping
[22:01:24] [Server thread/INFO] [JourneyMapServer]: On Request: Sending WorldID packet to: mamiemru
[22:01:24] [Server thread/INFO] [JourneyMapServer]: mamiemru is an EntityPlayerMP attempting to send the worldId packet
[22:01:42] [Server thread/ERROR] [FML]: There was a critical exception handling a packet on channel NEI
java.lang.NoClassDefFoundError: net/minecraft/client/resources/I18n
        at Launch//gtneioreplugin.util.DimensionHelper.getDimLocalizedName(DimensionHelper.java:325) ~[DimensionHelper.class:?]
        at Launch//gtneioreplugin.plugin.item.ItemDimensionDisplay.func_77653_i(ItemDimensionDisplay.java:63) ~[ItemDimensionDisplay.class:?]
        at Launch//net.minecraft.item.ItemStack.func_82833_r(ItemStack.java:427) ~[add.class:?]
        at Launch//net.minecraft.item.ItemStack.func_151000_E(ItemStack.java:759) ~[add.class:?]
        at Launch//codechicken.nei.NEIServerUtils.givePlayerItem(NEIServerUtils.java:283) ~[NEIServerUtils.class:?]
        at Launch//codechicken.nei.NEISPH.handleGiveItem(NEISPH.java:153) ~[NEISPH.class:?]
        at Launch//codechicken.nei.NEISPH.handlePacket(NEISPH.java:35) ~[NEISPH.class:?]
        at Launch//codechicken.lib.packet.PacketCustom$ServerInboundHandler.handle(PacketCustom.java:119) ~[PacketCustom$ServerInboundHandler.class:?]
        at Launch//codechicken.lib.packet.PacketCustom$CustomInboundHandler.channelRead0(PacketCustom.java:81) ~[PacketCustom$CustomInboundHandler.class:?]
        at Launch//codechicken.lib.packet.PacketCustom$CustomInboundHandler.channelRead0(PacketCustom.java:68) ~[PacketCustom$CustomInboundHandler.class:?]
        at Launch//io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:98) ~[minecraft_server.1.7.10.jar:?]
        at Launch//io.netty.channel.DefaultChannelHandlerContext.invokeChannelRead(DefaultChannelHandlerContext.java:337) ~[minecraft_server.1.7.10.jar:?]
        at Launch//io.netty.channel.DefaultChannelHandlerContext.fireChannelRead(DefaultChannelHandlerContext.java:323) ~[minecraft_server.1.7.10.jar:?]
        at Launch//io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:785) ~[minecraft_server.1.7.10.jar:?]
        at Launch//io.netty.channel.embedded.EmbeddedChannel.writeInbound(EmbeddedChannel.java:169) ~[minecraft_server.1.7.10.jar:?]
        at Launch//cpw.mods.fml.common.network.internal.FMLProxyPacket.func_148833_a(FMLProxyPacket.java:77) [FMLProxyPacket.class:?]
        at Launch//net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:212) [ej.class:?]
        at Launch//net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:165) [nc.class:?]
        at Launch//net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:659) [MinecraftServer.class:?]
        at Launch//net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:334) [lt.class:?]
        at Launch//net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:547) [MinecraftServer.class:?]
        at Launch//net.minecraft.server.MinecraftServer.run(MinecraftServer.java:427) [MinecraftServer.class:?]
        at Launch//net.minecraft.server.MinecraftServer$2.run(MinecraftServer.java:685) [li.class:?]
Caused by: java.lang.ClassNotFoundException: Class bytes are null for net.minecraft.client.resources.I18n (net.minecraft.client.resources.I18n, net.minecraft.client.resources.I18n)
        ... 23 more
[22:01:42] [Server thread/INFO]: mamiemru lost connection: TextComponent{text='A fatal error has occured, this connection is terminated', siblings=[], style=Style{hasParent=false, color=null, bold=null, italic=null, underlined=null, obfuscated=null, clickEvent=null, hoverEvent=null}}


### Your Proposal

don't make the session end.

### Final Checklist

- [x] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [x] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [ ] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

--- a/src/main/java/gtneioreplugin/plugin/item/ItemDimensionDisplay.java
+++ b/src/main/java/gtneioreplugin/plugin/item/ItemDimensionDisplay.java
@@ -1,9 +1,9 @@
 package gtneioreplugin.plugin.item;
 
 import net.minecraft.block.Block;
-import net.minecraft.client.resources.I18n;
 import net.minecraft.item.ItemBlock;
 import net.minecraft.item.ItemStack;
+import net.minecraft.util.StatCollector;
 import net.minecraftforge.client.MinecraftForgeClient;
 
 import cpw.mods.fml.common.FMLCommonHandler;
@@ -62,6 +62,6 @@ public class ItemDimensionDisplay extends ItemBlock {
 
         String i18nName = DimensionHelper.getDimLocalizedName(dimName);
 
-        return I18n.format(DimensionHelper.getDimTier(dimName), i18nName);
+        return StatCollector.translateToLocalFormatted(DimensionHelper.getDimTier(dimName), i18nName);
     }
 }

--- a/src/main/java/gtneioreplugin/util/DimensionHelper.java
+++ b/src/main/java/gtneioreplugin/util/DimensionHelper.java
@@ -53,7 +53,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
-import net.minecraft.client.resources.I18n;
+import net.minecraft.util.StatCollector;
 
 import com.github.bsideup.jabel.Desugar;
 
@@ -240,6 +240,6 @@ public class DimensionHelper {
     }
 
     public static String getDimLocalizedName(String dimName) {
-        return I18n.format(getDimUnlocalizedName(dimName));
+        return StatCollector.translateToLocal(getDimUnlocalizedName(dimName));
     }
 }

